docs/protocol/core/v3.0.rst

```diff
@@ -103,11 +103,72 @@ Data types
 
 TODO define core data types
 
-Regular chunk grids
--------------------
-
-TODO define regular chunk grids, including how to form a key for each chunk in a grid
-
+Chunk grids
+-----------
+
+A chunk grid defines a set of chunks which contain the elements of an
+array. The chunks of a grid form a tessellation of the array space,
+which is a space defined by the dimensionality and shape of the
+array. This means that every element of the array is a member of one
+chunk, and there are no gaps or overlaps between chunks.
+
+In general there are different possible types of grids. The core
+protocol defines the regular grid type, where all chunks are
+hyperrectangles of the same shape. Protocol extensions may define
+other grid types, such as rectilinear grids where chunks are still
+hyperrectangles but do not all share the same shape.
+
+A grid type also defines rules for constructing a unique key for each
+chunk, which is a string of ASCII characters that can be used to save
```

Comment:
This is perhaps for a different PR, but I could see specifying a "tuple of strings of ASCII characters in the set [...]". See below.